public/js/selectors.js

```diff
@@ -106,6 +113,24 @@ function makeLocationId(location: Location) {
   return location.sourceId + ":" + location.line.toString();
 }
 
+function getOriginalLocation(state: AppState, location) {
```

Comment:
Oh, here is `getOriginalLocation`. I feel like this and `getGeneratedLocation` should be in the same place.

This looks good though.